plugin/trino-clickhouse/src/test/java/io/trino/plugin/clickhouse/TestClickHouseConnectorTest.java

```diff
@@ -817,4 +922,202 @@ private Map<String, String> getTableProperties(String schemaName, String tableNa
+    {
+        return Optional.of(setup);
+    }
+
+    private List<SetColumnTypeSetup> setColumnTypeSetupData()
+    {
+        return ImmutableList.<SetColumnTypeSetup>builder()
+                .add(new SetColumnTypeSetup("tinyint", "TINYINT '127'", "smallint"))
+                .add(new SetColumnTypeSetup("smallint", "SMALLINT '32767'", "integer"))
+                .add(new SetColumnTypeSetup("integer", "2147483647", "bigint"))
+                .add(new SetColumnTypeSetup("bigint", "BIGINT '-2147483648'", "integer"))
+                .add(new SetColumnTypeSetup("real", "REAL '10.3'", "double"))
+                .add(new SetColumnTypeSetup("real", "REAL 'NaN'", "double"))
+                .add(new SetColumnTypeSetup("decimal(5,3)", "12.345", "decimal(10,3)")) // short decimal -> short decimal
+                .add(new SetColumnTypeSetup("decimal(28,3)", "12.345", "decimal(38,3)")) // long decimal -> long decimal
+                .add(new SetColumnTypeSetup("decimal(5,3)", "12.345", "decimal(38,3)")) // short decimal -> long decimal
+                .add(new SetColumnTypeSetup("decimal(5,3)", "12.340", "decimal(5,2)"))
+                .add(new SetColumnTypeSetup("decimal(5,3)", "12.35", "decimal(5,2)"))
+                .add(new SetColumnTypeSetup("varchar(100)", "'shorten-varchar'", "varchar"))
+                .add(new SetColumnTypeSetup("char(25)", "'shorten-char'", "varchar"))
+                .add(new SetColumnTypeSetup("char(20)", "'char-to-varchar'", "varchar"))
+                .add(new SetColumnTypeSetup("varchar", "'varchar-to-char'", "varchar"))
+                .build();
+    }
+
+    @Test(dataProvider = "setColumnTypesDataProvider")
+    @Override
+    public void testSetColumnTypes(SetColumnTypeSetup setup)
+    {
+        skipTestUnless(hasBehavior(SUPPORTS_SET_COLUMN_TYPE) && hasBehavior(SUPPORTS_CREATE_TABLE_WITH_DATA));
+
+        String tableName = "test_set_column_type_" + System.currentTimeMillis();
+        assertUpdate("CREATE TABLE " + tableName + "(col " + setup.sourceColumnType() + ", col2 int not null)  WITH (engine='mergetree', order_by=ARRAY['col2'])");
+        query("insert into " + tableName + "(col, col2)  values(CAST(" + setup.sourceValueLiteral() + " AS " + setup.sourceColumnType() + "), 2)");
+        Runnable setColumnType = () -> assertUpdate("ALTER TABLE " + tableName + " ALTER COLUMN col SET DATA TYPE " + setup.newColumnType());
+        if (setup.unsupportedType()) {
+            assertThatThrownBy(setColumnType::run)
+                    .satisfies(this::verifySetColumnTypeFailurePermissible);
+            return;
+        }
+        setColumnType.run();
+
+        assertEquals(getColumnType(tableName, "col"), setup.newColumnType());
+        assertThat(query("SELECT col FROM " + tableName))
+                .skippingTypesCheck()
+                .matches("SELECT " + setup.newValueLiteral());
+    }
+
+    @Test
+    @Override
+    public void testAddNotNullColumnToEmptyTable()
```

Comment:
Revert changes for testAddNotNullColumnToEmptyTable & testAddNotNullColumn.

Reply:
Hi @ebyhr , both the tests have to be overridden because it fails for the default Log Engine.

Reply:
I knew. However, that is unrelated to the commit for changing column types.